examples/lines_bars_and_markers/gradient_bar.py

```diff
@@ -10,12 +24,44 @@
 np.random.seed(19680801)
 
 
-def gbar(ax, x, y, width=0.5, bottom=0):
-    X = [[.6, .6], [.7, .7]]
+def gradient_image(ax, extent, direction=0.3, cmap_range=(0, 1), **kwargs):
+    """
+    Draw a gradient image based on a colormap.
+
+    Parameters
+    ----------
+    ax : Axes
+        The axes to draw on.
+    extent
+        The extent of the image as (xmin, xmax, ymin, ymax).
+        By default, this is in Axes coordinates but may be
+        changed using the *transform* kwarg.
+    direction : float
+        The direction of the gradient. This is a number in
+        range 0 (=vertical) to 1 (=horizontal).
```

Comment:
Just passing the angle (in degrees or in radians, up to you) seems simpler (in particular to explain)?

Reply:
It would be the angle only for square images. In general, it's a (nonlinear monotoneous) parametrization of the angle due to the distortion. I did not bother to untangle the transformation from `extent`; and I didn't want to raise false expectations.

If you want to make real angles work, you are welcome to provide a PR. OTOH, maybe one does not want real angles - probably depends on the application. 0.5 is now always a gradient along the diagonal of the image.

Reply:
Good point, let's just leave it as it is.